caikit_nlp/modules/embedding_retrieval/embedding.py

```diff
@@ -0,0 +1,140 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from pathlib import Path
+from typing import List
+import os
+
+# Third Party
+from sentence_transformers import SentenceTransformer
+
+# First Party
+from caikit.core import ModuleBase, ModuleConfig, ModuleSaver, module
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .embedding_retrieval_task import EmbeddingRetrievalTask
+from caikit_nlp.data_model.embedding_vectors import EmbeddingResult, Vector1D
+
+logger = alog.use_channel("<EMBD_BLK>")
+error = error_handler.get(logger)
+
+HOME = Path.home()
+DEFAULT_HF_MODEL = "sentence-transformers/all-MiniLM-L6-v2"
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingRetrievalTask,
+)
+class EmbeddingModule(ModuleBase):
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str) -> "EmbeddingModule":
+        """Load a sequence classification model
+
+        Args:
+            model_path: str
+                Path to the config dir of the model to be loaded.
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        load_path = config.get("model_artifacts")
+
+        artifact_path = False
+        if load_path:
+            if os.path.isabs(load_path) and os.path.isdir(load_path):
+                artifact_path = load_path
+            else:
+                full_path = os.path.join(model_path, load_path)
+                if os.path.isdir(full_path):
+                    artifact_path = full_path
+
+        if not artifact_path:
+            artifact_path = config.get("hf_model", DEFAULT_HF_MODEL)
+
+        return cls.bootstrap(artifact_path)
+
+    def run(
+        self, input: List[str], **kwargs  # pylint: disable=redefined-builtin
```

Comment:
Usual paradigm is for `.run` function to accept `str` and `run_batch` function to accept `List[str]` can we split this into 2 functions ?

Reply:
We'll have to discuss this.  It would be nice to have a str | List[str], but also the ability to send one gRPC/REST request with a bunch of inputs.

Reply:
I think this is good as-is until we get multi-transaction way of doing both here.

Reply:
Created follow-up issue [here](https://github.com/caikit/caikit/issues/527)

Reply:
latest version is single str only.  Will work on multi-task (the multi-"transaction" mis-speak is my bad) in separate PR.